openedx/core/djangoapps/content/learning_sequences/api/processors/base.py

```diff
@@ -43,6 +43,7 @@ def __init__(self, course_key: CourseKey, user: types.User, at_time: datetime):
         self.course_key = course_key
         self.user = user
         self.at_time = at_time
+        self.course_version = course_version
```

Comment:
So I had another issue where I wanted more data related to the course when initializing a processor, back when I was trying to generalize the `EnrollmentTrackOutlineProcessor` to handle any user partition types–because I wanted to only load the user partitions that actually existed in the outline of the content (and not all the A/B tests that were in Units). That would have required introspection of the course outline content before `load_data()` finished executing.

So instead of adding `course_version` as a param here, how would you feel about giving `load_data` an argument that takes a whole `CourseOutline` (like `usage_keys_remove` and `inaccessible_sequences` already do)? And then the processor can figure out what fields it wants to extract from that?

Reply:
This change makes a lot of sense, @ormsbee . I'll proceed with making it.

Reply:
I've made this change.